pinecone-io/pinecone-ts-client
# [v1] Add describeIndexStats method to new client

## Problem
The Node V1 client is currently missing functionality for supporting `/describe_index_stats`.

## Solution
- Add a new `describeIndexStats.ts` file under `/src/data/`.
- Implement `describeIndexStats` to support HTTPS calls to retrieve statistics for a single index.
- Add a new `describeIndexStats.test.ts` file to support unit tests.


## Type of Change


## Test Plan
New unit tests added for coverage of the basic functionality.

### Manual Testing
```
> npm run repl
> await init()

// upsert vectors and check the index stats
> await client.index('test-index').upsert([{id: '111', values: [1,2]}, {id: '222', values: [2,3]}, {id: '333', values: [3,4]}])

> await client.index('test-index').describeIndexStats()
{
  namespaces: { '': { vectorCount: 3 } },
  dimension: 2,
  indexFullness: 0,
  totalVectorCount: 3
}

// upsert vectors with metadata and leverage the filter field when querying stats
> await client.index('test-index').upsert([
  { id: '444', values: [1, 2], metadata: { genre: 'ambient' } },
  { id: '555', values: [2, 3], metadata: { genre: 'classical' } },
])

> await client.index('test-index').describeIndexStats({ filter: { genre: 'classical' }})
{
  namespaces: { '': { vectorCount: 1 } },
  dimension: 2,
  indexFullness: 0,
  totalVectorCount: 5
}
```
